app/src/main/java/protect/card_locker/LoyaltyCard.java

```diff
@@ -97,51 +229,185 @@ public void writeToParcel(Parcel parcel, int i) {
+        } else if (requireFull) {
+            throw new IllegalArgumentException("Missing key " + BUNDLE_LOYALTY_CARD_LAST_USED);
+        }
+        if (bundle.containsKey(BUNDLE_LOYALTY_CARD_ZOOM_LEVEL)) {
+            setZoomLevel(bundle.getInt(BUNDLE_LOYALTY_CARD_ZOOM_LEVEL));
+        } else if (requireFull) {
+            throw new IllegalArgumentException("Missing key " + BUNDLE_LOYALTY_CARD_ZOOM_LEVEL);
+        }
+        if (bundle.containsKey(BUNDLE_LOYALTY_CARD_ARCHIVE_STATUS)) {
+            setArchiveStatus(bundle.getInt(BUNDLE_LOYALTY_CARD_ARCHIVE_STATUS));
+        } else if (requireFull) {
+            throw new IllegalArgumentException("Missing key " + BUNDLE_LOYALTY_CARD_ARCHIVE_STATUS);
+        }
+    }
 
-        if (cursor.isNull(balanceTypeColumn) == false) {
-            balanceType = Currency.getInstance(cursor.getString(balanceTypeColumn));
+    public Bundle toBundle() {
+        Bundle bundle = new Bundle();
+
+        bundle.putInt(BUNDLE_LOYALTY_CARD_ID, id);
+        bundle.putString(BUNDLE_LOYALTY_CARD_STORE, store);
+        bundle.putString(BUNDLE_LOYALTY_CARD_NOTE, note);
+        if (validFrom != null) {
+            bundle.putLong(BUNDLE_LOYALTY_CARD_VALID_FROM, validFrom.getTime());
+        }
+        if (expiry != null) {
+            bundle.putLong(BUNDLE_LOYALTY_CARD_EXPIRY, expiry.getTime());
+        }
+        bundle.putString(BUNDLE_LOYALTY_CARD_BALANCE, balance.toString());
+        if (balanceType != null) {
+            bundle.putString(BUNDLE_LOYALTY_CARD_BALANCE_TYPE, balanceType.toString());
+        }
+        bundle.putString(BUNDLE_LOYALTY_CARD_CARD_ID, cardId);
+        bundle.putString(BUNDLE_LOYALTY_CARD_BARCODE_ID, barcodeId);
+        if (barcodeType != null) {
+            bundle.putString(BUNDLE_LOYALTY_CARD_BARCODE_TYPE, barcodeType.name());
+        }
+        if (headerColor != null) {
+            bundle.putInt(BUNDLE_LOYALTY_CARD_HEADER_COLOR, headerColor);
         }
+        bundle.putInt(BUNDLE_LOYALTY_CARD_STAR_STATUS, starStatus);
+        bundle.putLong(BUNDLE_LOYALTY_CARD_LAST_USED, lastUsed);
+        bundle.putInt(BUNDLE_LOYALTY_CARD_ZOOM_LEVEL, zoomLevel);
+        bundle.putInt(BUNDLE_LOYALTY_CARD_ARCHIVE_STATUS, archiveStatus);
 
+        return bundle;
+    }
+
+    public static LoyaltyCard fromCursor(Cursor cursor) {
+        // id
+        int id = cursor.getInt(cursor.getColumnIndexOrThrow(DBHelper.LoyaltyCardDbIds.ID));
+        // store
+        String store = cursor.getString(cursor.getColumnIndexOrThrow(DBHelper.LoyaltyCardDbIds.STORE));
+        // note
+        String note = cursor.getString(cursor.getColumnIndexOrThrow(DBHelper.LoyaltyCardDbIds.NOTE));
+        // validFrom
+        Date validFrom = null;
+        long validFromLong = cursor.getLong(cursor.getColumnIndexOrThrow(DBHelper.LoyaltyCardDbIds.VALID_FROM));
         if (validFromLong > 0) {
             validFrom = new Date(validFromLong);
         }
```

Comment:
Just a suggestion, but this seems cleaner to me (also for the other ones below):

```suggestion
        long validFromLong = cursor.getLong(cursor.getColumnIndexOrThrow(DBHelper.LoyaltyCardDbIds.VALID_FROM));
        Date validFrom = validFromLong > 0 ? new Date(validFromLong) : null;
```